### ISSUE TITLE
[camelcase] ignore non-camelcase properties required by external lib's typings

### ISSUE THREAD

**Issue by quezak** [typescript-eslint/typescript-eslint#1600]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/camelcase": "error"
  }
}
```

```TS
// a real example from validator.js
import { isURL } from 'validator';
// ...
isURL('somestring', { require_tld: false });  // Error: ... is not in camel case

// also a real example from an actual npm package, just stripped the company name ;)
import { SomeApiClient } from 'somecompany-client';  
// ...
new SomeApiClient({ client_id: 'xxx', client_secret: 'yyy' });  // Error: ... is not in camel case
```

**Expected Result**
No errors when the offending property names are defined by an external (ie. imported from `node_modules`) library.

**Actual Result**
I need to disable the rule or whitelist properties each time I'm using such libraries.

**Additional Info**
Please let me know, if this is a general problem in eslint and I should file the issue there.
But I don't think it is -- eslint doesn't have type info, so it can't determine that the offending non-camelcase name is required by the external library, can it?

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.19.2` |
| `@typescript-eslint/parser`        | `2.19.2` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `11.9` |
| `npm`                              | `6.5` |

(but I don't use `npm`, only `yarn` ^)

**Comment by bradzacher** [comment]
This is just a general bug with the eslint rule.
Suggest asking eslint core to add a new option to not check property names if the object is not part of a variable declaration (i.e. are part of a call expression). Though note that this is all open source, so you'd probably be best to raise an issue to discuss with the eslint core maintainers, and then look to submitting a PR for your case if they accept it.

----

We will not be adding the additional type-based logic to the camelcase rule to support this use case.
This will require bringing type info into the rule, which will exclude many of our userbase from using the rule - many users do not use type information because of the perf implications.
